packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts

```diff
@@ -130,6 +130,27 @@ const processOnEvent = async (event: AWSCDKAsyncCustomResource.OnEventRequest):
   let result;
   switch (event.RequestType) {
     case 'Create':
+      if (tableDef.isImported) {
```

Comment:
1. rather than inlining this big `if` block, can we refactor to a standalone function to make the "Create" case a bit easier to parse?
2. I don't see anything to suggest otherwise, but please confirm that this won't cause any issues with the Tagging support for Amplify-managed tables that we're introducing in https://github.com/aws-amplify/amplify-category-api/pull/2694

Reply:
1. Will do.
2. I don't see any issues here, but I'll add a test to cover this case before merging to main.